src/state/routing/useRoutingAPITrade.ts

```diff
@@ -77,8 +78,11 @@ export function useRoutingAPITrade<TTradeType extends TradeType>(
   const isCurrent = currentTradeResult === tradeResult
 
   return useMemo(() => {
-    if (!amountSpecified || isError || !queryArgs) {
-      return { state: TradeState.INVALID, trade: tradeResult?.trade }
+    if (!shouldGetTrade && amountSpecified) {
+      // If we don't want to fetch new trades, but have valid inputs, return the stale trade.
```

Comment:
`shouldGetTrade` means we aren't polling new trades, but if the input is valid we still want to return the stale trade. for the UI to display.